### PR TITLE
Link login modal questions

### LINKED ISSUE #4159: Link Login Modal to Flag button
We are creating a login/signup modal, next step is to link to various pages. In this issue we will be linking it to Questions page
publiclab.org/questions/
app/views/questions/show.html.erb
image

### PR DESCRIPTION
**Pull request by okonek**
Fixes #4159  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

spamquestions

### CODE REVIEW

**Comment by okonek** [comment]
@SidharthBansal Is it ok? When the user is not logged in and he clicks the spam button the login modal shows up and then redirects.

**Comment by okonek** [comment]
I claimed it: https://codein.withgoogle.com/dashboard/task-instances/4522206253023232/

**Comment by okonek** [comment]
Thank you!

**Comment by SidharthBansal** [comment]
@jywarren please review and merge this.

**Comment by jywarren** [comment]
I think it's starting to take up a lot of space here -- and we'd like to have the "flag" associated with the "spam" -- what if we had a single Flag button which opens up a Bootstrap popover, which shows "Flag" this to moderators (email) and "Spam" this if you are a moderator (and link to learn more about moderators). These could fit under just one tiny flag icon, and expand out when you click it. 

However, you've successfully completed the original task even if this isn't merged yet -- so you should get approved for the task! Thanks!

**Comment by SidharthBansal** [comment]
I have already rewarded him the points

**Comment by SidharthBansal** [comment]
>I think it's starting to take up a lot of space here -- and we'd like to have the "flag" associated with the "spam" -- what if we had a single Flag button which opens up a Bootstrap popover, which shows "Flag" this to moderators (email) and "Spam" this if you are a moderator (and link to learn more about moderators). These could fit under just one tiny flag icon, and expand out when you click it.

I think we should merge this now. And open up a different issue for the above comment. We can brainstorm on it. What do you say?

**Comment by jywarren** [comment]
It's OK - we can just leave it open with the task approved... and solve this in follow-up in the same PR. Sound OK?

**Comment by SidharthBansal** [comment]
Sounds great!!!

**Comment by jywarren** [comment]
OK - thanks for your patience - how about this as a solution: 

image


where the top has a flag icon, the two middle ones have the ban (like circle with slash) icon, and the bottom is a link to /moderators

(mockup made [in this doc](https://docs.google.com/presentation/d/1TCZoTfuhamRVrUak8aDgqJAwSgyhRtZg2Pgacl2_4zc/edit#slide=id.g4877128470_0_0))

**Comment by jywarren** [comment]
The ones that say "only mods and admins" could be a link instead that says "Log in to moderate" maybe... but really you'd still have to become a mod or admin to be able to so I don't know if it's necessary?

And since we have these flag icons in so many places, maybe we should think about making this a partial template or something standard (as with the login modal) we can insert anywhere?

**Comment by SidharthBansal** [comment]
Oh it is so nice.
It is compact design and also reduced redundancy and ambiguity.
I really loved it
We can first complete this issue. Once we complete this issue we can raise an issue to identify these flag icons at different places throughout the website. Then we can create ftos for linking flag buttons to partials and for deletion of the redundant buttons.

**Comment by gauravano** [comment]
Hi @okonek would you like to implement the design suggested by @jywarren ? Thanks!

File: app/views/questions/_questions.html.erb [hunk 1]
```diff
@@ -22,7 +22,12 @@
           <span id="tag_<%= tag.id %>" class="label label-primary"><a href="/questions/tag/<%= tag.name %>"><%= tag.name %></a></span>
         <% end %>
         </span>
-        <p style="margin-top:30px;"><% if logged_in_as(['admin','moderator']) %><a class="btn btn-default btn-xs" href="/moderate/spam/<%= node.id %>"><i class="fa fa-ban"></i> Spam</a><% end %>
+        <p style="margin-top:30px;">
```
Comment (SidharthBansal) [inline app/views/questions/_questions.html.erb]: Due to recent changes by Kevin and design suggested by Jeff, we need to change this code a lot. So, I am closing this PR. @okonek please feel free to open up a PR for the design suggested by Jeff () and then a follow up PR for the linking of modals(#4255).

Comment (SidharthBansal) [reply app/views/questions/_questions.html.erb]: These changes are no longer needed due to new PRs merged.

Comment (SidharthBansal) [reply app/views/questions/_questions.html.erb]: Please feel free to open a new pr @okonek 
Thanks for awesome work you are doing at PL

**Pull request closed without merging**